src/queue/QueueLogBehavior.php

```diff
@@ -86,29 +93,15 @@ public function afterError(ExecEvent $event): void
     }
 
     /**
-     * Changes the file that logs will get flushed to.
+     * Enables the log target logs will get flushed to.
      */
-    private function _changeLogFile(): void
+    private function _enableLogTarget(): void
     {
-        $logDispatcher = Craft::$app->getLog();
-
-        foreach ($logDispatcher->targets as $target) {
-            if ($target instanceof FileTarget) {
-                // Log to queue.log
-                $target->logFile = Craft::getAlias('@storage/logs/queue.log');
-
-                // Don't log global vars
-                $target->logVars = [];
-
-                // Prevent verbose system logs
-                if (!YII_DEBUG) {
-                    $target->except = ['yii\*'];
-                    $target->setLevels(['info', 'warning', 'error']);
-                }
-
-                break;
-            }
-        }
+        Collection::make(Craft::$app->getLog()->targets)
+            ->whereInstanceOf(MonologTarget::class)
+            ->each(function(MonologTarget $target) {
+                $target->enabled = $target->name === Dispatcher::TARGET_QUEUE;
+            });
```

Comment:
Instead of swapping the filename, we enable the queue target and disable the others.